ReactViewResources/Loader.ts

```diff
@@ -0,0 +1,351 @@
+﻿declare const CefSharp: {
+    BindObjectAsync(objName1: string, objName2: string): Promise<void>
+};
+
+type Dictionary<T> = { [key: string]: T };
+
+// TODO missing typings
+type React = any;
+type ReactDOM = any;
+
+class Task<ResultType> {
+
+    private taskPromise: Promise<ResultType>;
+    private resolve: (result: ResultType) => void;
+
+    constructor() {
+        this.taskPromise = new Promise<ResultType>((resolve) => this.resolve = resolve);
+    }
+
+    public setResult(result?: ResultType) {
+        this.resolve(result as ResultType);
+    }
+
+    public get promise() {
+        return this.taskPromise;
+    }
+}
+
+const [UseEnhancedRenderingEngine, EnableDebugMode, LibsPath, ModulesObjectName, EventListenerObjectName, ReadyEventName] = Array.from(new URLSearchParams(location.search).keys());
+
+const Modules: Dictionary<{}> = {};
+window[ModulesObjectName] = Modules;
+
+const StylesheetsLoadTask = new Task();
+const PluginsLoadTask = new Task();
+const BootstrapTask = new Task();
+
+export function showErrorMessage(msg: string): void {
+    const ContainerId = "webview_error";
+    let msgContainer = document.getElementById(ContainerId) as HTMLDivElement;
+    if (!msgContainer) {
+        msgContainer = document.createElement("div");
+        msgContainer.id = ContainerId;
+        let style = msgContainer.style;
+        style.backgroundColor = "#f45642";
+        style.color = "white";
+        style.fontFamily = "Arial";
+        style.fontWeight = "bold";
+        style.fontSize = "10px"
+        style.padding = "3px";
+        style.position = "absolute";
+        style.top = "0";
+        style.left = "0";
+        style.right = "0";
+        style.zIndex = "10000";
+        style.height = "auto";
+        style.wordWrap = "break-word";
+        document.body.appendChild(msgContainer);
+    }
+    msgContainer.innerText = msg;
+}
+
+function loadRequire(): Promise<void> {
+    return new Promise((resolve) => {
+        // load require js
+        let requireScript = document.createElement("script");
+        requireScript.src = LibsPath + "requirejs/require.js";
+        requireScript.addEventListener("load", () => resolve());
+        if (document.head) {
+            document.head.appendChild(requireScript);
+        }
+    });
+}
+
+function loadFramework(): void {
+    if (UseEnhancedRenderingEngine === "1") {
```

Comment:
In case of using preact, we're loading both preact and react (considering that there is no "else")?